### PR TITLE
perf: drop deps to save 4372 bytes gzipped

### PR DESCRIPTION
**Pull request by brandonocasey**

### CODE REVIEW

**Comment by brandonocasey** [comment]
I am still wondering if we want to get rid of `is-function`, even though it's small. It's still overhead and a dependency I don't think we need. We is `typeof x === 'function'`  all over the place in our codebase.

**Comment by gkatsev** [comment]
There is a risk with removing is-function, though. Give that we don't know if people are using our XHR helper since it's exposed. If they are doing something where is-function returns true but just a check with typeof doesn't, it'll start failing. I think it's better to reduce the risk here by keeping is-function, considering it's size.

File: index.js [hunk 1]
```diff
@@ -1,8 +1,39 @@
 "use strict";
 var window = require("global/window")
-var isFunction = require("is-function")
-var parseHeaders = require("parse-headers")
-var xtend = require("xtend")
+var _extends = require("@babel/runtime/helpers/extends");
```
Comment (gkatsev) [inline index.js:3]: Do we run XHR through babel? I'm not sure this would get deduped properly.

Comment (brandonocasey) [reply index.js:3]: We run it through rollup, and rollup will dedupe it for us

**Review by gkatsev — APPROVED** [review]
Did we want to move this repo to the videojs and publish from there?

File: index.js (hunk 1 above)
Comment (brandonocasey) [reply index.js:3]: babel just adds in the `import`/`require` statement for us using `@babel/plugin-runtime-transform` so we will have a duplicate dependency and rollup will see that and dedupe them.

**Pull request merged**